src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java

```diff
@@ -225,6 +225,51 @@ public Dataset find(Object pk) {
         return typedQuery.getResultList();
     }
     
+        /**
+     * For docs, see the equivalent method on the DataverseServiceBean.
+     * @param numPartitions
+     * @param partitionId
+     * @param skipIndexed
+     * @return a list of datasets
+     * @see DataverseServiceBean#findAllOrSubset(long, long, boolean)
+     */     
+    public List<Long> findAllOrSubsetOrderByFilesOwned(boolean skipIndexed) {
+        /*
+        Disregards deleted or replaced files when determining 'size' of dataset.
```

Comment:
BTW, I am definitely OK with using the raw number of files for the ordering. We know that it's not a 100% accurate predictor of speed (both because of potentially replaced or deleted files; and because of drafts vs. published versions). However, our goal is not to be super accurate. But to be able to index the bulk of the database faster, by delaying indexing of a few outlier cases. That strategy appears to work. Even if some of these outliers don't actually take hours to index. 
... And I'm very happy with the efficient implementation of the actual sorting - by a native query, all on the database side and not requiring any instantiations on the database side.